test/src/androidTest/kotlin/io/realm/shared/notifications/RealmListNotificationsTests.kt

```diff
@@ -33,38 +47,208 @@ class RealmListNotificationsTests : NotificationTests {
+                    .observe()
+                    .collect { flowList ->
+                        channel1.trySend(flowList)
+                    }
+            }
+            val observer2 = async {
+                managedContainer.objectListField
+                    .observe()
+                    .collect { flowList ->
+                        channel2.trySend(flowList)
+                    }
+            }
+
+            // Ignore first emission with empty lists
+            channel1.receive()
+            channel2.receive()
+
+            // Trigger an update
+            realm.write {
+                val objects = objects<RealmListContainer>()
+                val queriedContainer = objects
+                    .first()
+                queriedContainer.objectListField
+                    .addAll(values.map { copyToRealm(it) })
+            }
+            assertEquals(OBJECT_VALUES.size, channel1.receive().size)
+            assertEquals(OBJECT_VALUES.size, channel2.receive().size)
+
+            // Cancel observer 1
+            observer1.cancel()
+
+            // Trigger another update
+            realm.write {
+                val queriedContainer = objects<RealmListContainer>()
+                    .first()
+                queriedContainer.objectListField
+                    .add(copyToRealm(RealmListContainer().apply { stringField = "C" }))
+            }
+
+            // Check channel 1 didn't receive the update
+            assertEquals(OBJECT_VALUES.size + 1, channel2.receive().size)
+            assertTrue(channel1.isEmpty)
+
+            observer2.cancel()
+            channel1.close()
+            channel2.close()
+        }
     }
 
     @Test
-    @Ignore
     override fun deleteObservable() {
-        TODO("Waiting for RealmList support")
+        runBlocking {
+            // Freeze values since native complains if we reference a package-level defined variable
+            // inside a write block
+            val values = OBJECT_VALUES.freeze()
+            val channel = Channel<RealmList<*>>(capacity = 1)
+            val managedContainer = realm.write {
+                RealmListContainer()
+                    .apply {
+                        objectListField.addAll(values.map { copyToRealm(it) })
+                    }.let { container ->
+                        copyToRealm(container)
+                    }
+            }
+            val observer = async {
+                managedContainer.objectListField
+                    .observe()
+                    .collect { flowList ->
+                        channel.send(flowList)
+                    }
+            }
+
+            // Assert container got populated correctly
+            val emittedList = channel.receive()
+            assertNotNull(emittedList)
+            assertEquals(OBJECT_VALUES.size, emittedList.size)
+
+            realm.write {
+                delete(findLatest(managedContainer)!!)
+            }
+
```

Comment:
What are we asserting here? Can't recall whether we aligned on just closing it (assert something about no events) or actually completing it (assert that it is completed with onCompletion).

Reply:
My assumption is that given the owner object is deleted the list will not be valid anymore and therefore shouldn't be emitted at all - since we emit frozen lists, freezing an invalid list would throw, so there is no assertion on anything. I guess that makes sense...?

Reply:
But isn't it a bit like https://github.com/realm/realm-kotlin/blob/master/test/src/androidTest/kotlin/io/realm/shared/notifications/RealmObjectNotificationsTests.kt#L128, where deletion is signalled by _completion_? @cmelchior, would it make sense to complete the flow for list on deletion of an object too?